### PR TITLE
Pull request to integrate weather App features into development

### PR DESCRIPTION
**Pull request by gango-anan**
# In this project, I:
- Signed up for an account with OpenWeatherMap and got the API key.
- Setup the working environment, i.e initialized npm, installed and configured webpack, configured linters for both the local and remote environment.
- Setup the skeleton structure of the application using HTML.
- Styled the application using Saas.
- Added the functionality to retrieve the weather data using the API.
- Added functionality to render the retrieved data.
- Added the functionality to convert temperature from the default scale( kelvin) to Fahrenheit and celsius.
- Added the functionality to display errors page when a city is not found.
- Added functionality to toggle between the temperature scales.
- Refactored code to remove unused code and fix errors raised by linters.

### CODE REVIEW

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light:

Hi @gango-anan, great job on this project :smile: The application works as expected :fireworks: Webpack configured correctly :+1: Async and await utilized:smile: Toggle functionality implemented :1st_place_medal: Linters set up correctly :100: There is one missing requirement that needs to be addressed, kindly check the comment below:

- Kindly include the functionality where the appearance changes based on the received data

Once this is resolved, kindly submit another code review request :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

File: src/render.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      weatherDetailsContainer.innerHTML = `
+      <p class="current-date">${nowDate.toLocaleTimeString()} , ${nowDate.toDateString()}</p>
+      <h2>${data.city}, ${codeToCountry.countryName(data.country)}</h2>
+      <p>Feels like ${displayedTemp(data.temperature)}${tempType()}</p>
+      <p>Humidity: ${data.humidity}%</p>
+      <p>Wind Speed: ${data.windSpeed} m/s</p>
+      <p>Visibility: ${convertVisibility(data.visibility)} km</p>`;
+      weatherIconContainer.innerHTML = `<img src="http://openweathermap.org/img/wn/${data.icon}@2x.png" alt="weather-icon"></img>`;
+      if (data.city.toLowerCase() !== 'kampala' && data.status.toLowerCase() === 'rain') {
+        weatherFocus.style.backgroundImage = "url('./images/rainy.gif')";
```
Comment (rindrajosia) [inline src/render.js]: Keep all the styling logic in CSS files

File: src/render.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      } else if (data.city.toLowerCase() !== 'kampala' && data.status.toLowerCase() === 'clouds') {
+        weatherFocus.style.backgroundImage = "url('./images/cloudy.gif')";
```
Comment (rindrajosia) [inline src/render.js]: Keep all the styling logic in CSS files

File: src/render.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      } else if (data.city.toLowerCase() !== 'kampala' && data.status.toLowerCase() === 'clear') {
+        weatherFocus.style.backgroundImage = "url('./images/clear.gif')";
```
Comment (rindrajosia) [inline src/render.js]: Keep all the styling logic in CSS files

**Review by rindrajosia — APPROVED** [review]
Approved! :tada:

Congratulations are in order!! :clap:  Keep up the good work!

- [ ] [OPTIONAL] Keep all the styling logic in CSS files (If you need to change something like a background image, add the classes with the different backgrounds to the CSS and then change the class name attribute with JavaScript) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/styling_requirement.png) from Pathwright)

All the best on your next projects! :wink:
Rindra Josia

File: src/render.js (hunk 1 above)
Comment (gango-anan) [reply src/render.js]: Noted and implemented, thanks for the recommendation.

File: src/render.js (hunk 2 above)
Comment (gango-anan) [reply src/render.js]: Noted and implemented, thanks for the recommendation.

File: src/render.js (hunk 3 above)
Comment (gango-anan) [reply src/render.js]: Noted and implemented, thanks for the recommendation.

**Pull request merged**